models/topics-model.js

```diff
@@ -39,35 +39,75 @@ exports.fetchArticleById = (id) => {
 };
 
 exports.fetchArticles = () => {
-  return db.query(`
+  return db
+    .query(
+      `
     SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comment_id) AS comment_count
     FROM articles
     LEFT JOIN comments ON comments.article_id = articles.article_id
     GROUP BY articles.article_id
     ORDER BY created_at DESC
-    `)
+    `
+    )
     .then((articles) => {
-
       const updatedArticles = articles.rows.map((article) => {
-      article.comment_count = Number(article.comment_count)
-      return article
-      })
-  
-      return updatedArticles
-    })
-}
+        article.comment_count = Number(article.comment_count);
+        return article;
+      });
+
+      return updatedArticles;
+    });
+};
 
 exports.fetchArticleIdComments = (id) => {
   return db
-  .query(
-    `
+    .query(
+      `
   SELECT * FROM comments
   WHERE article_id = $1
   ORDER BY created_at DESC
   `,
-    [id]
-  )
-  .then((comments) => {
-    return comments.rows
-  })
-}
\ No newline at end of file
+      [id]
+    )
+    .then((comments) => {
+      return comments.rows;
+    });
+};
+
+exports.addCommentToArticleId = (id, body, username) => {
+  if (typeof body === "number") {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+  }
+
+  return db
+    .query(
+      `
+  SELECT * FROM articles
+  WHERE article_id = $1
+  `,
+      [id]
+    )
+    .then((article) => {
+      if (article.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
+      return null;
+    })
+    .then(() => {
+    return db.query(
+        `
+   INSERT INTO comments
+   (body, article_id, author)
+   VALUES
+   ($1, $2, $3)
+   RETURNING *
+  `,
+        [body, id, username]
+      );
+    })
+    .then((addedComment) => {
+      addedComment.rows[0].created_at = String(addedComment.rows[0].created_at);
```

Comment:
Lets allow the database to dictate how the date comes back on the response. When we're asserting against it we can deal with it how it comes, it's probably fine to expect.any(String) for the data as the format is tricky to predict.